admin-console/src/main/java/com/agiletec/apsadmin/portal/PageFinderAction.java

```diff
@@ -216,12 +216,15 @@ public ITreeNode getAllowedTreeRootNode() {
 	}
 
 	private void addTreeWrapper(PageTreeNodeWrapper parent, IPage currentTreeNode) {
-		IPage[] children = currentTreeNode.getChildren();
+		boolean isOnline = currentTreeNode.isOnline();
+		String[] children = currentTreeNode.getChildrenCodes();
 		for (int i = 0; i < children.length; i++) {
-			IPage newCurrentTreeNode = children[i];
+			IPage newCurrentTreeNode = (isOnline)
+					? this.getPageManager().getOnlinePage(children[i])
+					: this.getPageManager().getDraftPage(children[i]);

```

Comment:
See above. Move this to a utility.

Reply:
See the reply for the class PageAction.java